nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/distribution/SimplifiedServerInfoWriter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management.distribution;
+
+import static org.nuxeo.ecm.core.io.registry.reflect.Instantiations.SINGLETON;
+import static org.nuxeo.ecm.core.io.registry.reflect.Priorities.REFERENCE;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import org.nuxeo.common.Environment;
+import org.nuxeo.ecm.admin.runtime.PlatformVersionHelper;
+import org.nuxeo.ecm.admin.runtime.SimplifiedBundleInfo;
+import org.nuxeo.ecm.admin.runtime.SimplifiedServerInfo;
+import org.nuxeo.ecm.core.io.marshallers.json.AbstractJsonWriter;
+import org.nuxeo.ecm.core.io.registry.reflect.Setup;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+
+@Setup(mode = SINGLETON, priority = REFERENCE)
+public class SimplifiedServerInfoWriter extends AbstractJsonWriter<SimplifiedServerInfo> {
+
+    @Override
+    public void write(SimplifiedServerInfo entity, JsonGenerator jg) throws IOException {
+        Collections.sort(entity.getBundleInfos());
+        jg.useDefaultPrettyPrinter();
+        jg.writeStartObject();
+        jg.writeStringField(Environment.PRODUCT_NAME, PlatformVersionHelper.getApplicationName());
+        jg.writeStringField(Environment.PRODUCT_VERSION, PlatformVersionHelper.getApplicationVersion());
+        jg.writeStringField(Environment.DISTRIBUTION_NAME, PlatformVersionHelper.getDistributionName());
+        jg.writeStringField(Environment.DISTRIBUTION_VERSION, PlatformVersionHelper.getDistributionVersion());
+        jg.writeStringField(Environment.DISTRIBUTION_SERVER, PlatformVersionHelper.getDistributionHost());
+        jg.writeStringField(Environment.DISTRIBUTION_DATE, PlatformVersionHelper.getDistributionDate());
```

Comment:
For each field, you shoud be using the `SimplifiedServerInfo` API that encapsulates the implementation, e.g. `SimplifiedServerInfo#getApplicationName`.

Reply:
You could also add the Warnings on Startup, as in the JSF UI.

Reply:
I added an Environment.WARNINGS constant so it can be used in both writer and test.

Reply:
Not sure it's a good idea, `Environment` is to describe Nuxeo's environment values such as the log or data dir, and some framework property keys. Here you just need a constant for the writer, it can be declared in the writer.
By the way, the commit doesn't have the new constant in Environment :smiley: